pandas/core/arrays/categorical.py

```diff
@@ -523,14 +523,17 @@ def astype(self, dtype: Dtype, copy: bool = True) -> ArrayLike:
             try:
                 new_cats = np.asarray(self.categories)
                 new_cats = new_cats.astype(dtype=dtype, copy=copy)
+                fill_value = np.array(np.nan).astype(dtype).item()
```

Comment:
there are cases where .item() has weird behavior and we use lib.item_from_zerodim instead.  no idea if any of those cases are reachable here